src/SFML/Window/Win32/JoystickImpl.cpp

```diff
@@ -45,1114 +45,717 @@
 
-    if (result != ERROR_SUCCESS)
+    if (jstkImpl.m_useXInput)
     {
-        sf::err() << "Unable to query registry key for joystick at index " << index << ": "
-                  << sf::priv::getErrorString(static_cast<DWORD>(result)) << std::endl;
-        return joystickDescription;
+        // XInput has 14 Buttons and 6 Axes
+        jstkImpl.m_caps.buttonCount = 14;
+        const auto axes             = 6;
+        for (auto index = 0; index < Joystick::AxisCount && index < axes; index++)
+        {
+            jstkImpl.m_caps.axes[GetAxis(index)] = true;
+        }
     }
-
-    subkey = REGSTR_PATH_JOYOEM;
-    subkey += TEXT('\\');
-    subkey.append(keyData.data(), keyDataSize / sizeof(TCHAR));
-
-    result = RegOpenKeyEx(rootKey, subkey.c_str(), 0, KEY_READ, &currentKey);
-
-    if (result != ERROR_SUCCESS)
+    else
     {
-        sf::err() << "Unable to open registry key for joystick at index " << index << ": "
-                  << sf::priv::getErrorString(static_cast<DWORD>(result)) << std::endl;
-        return joystickDescription;
+        // RawInput query capabilities
+        UINT buffSize = {0};
+        success       = GetRawInputDeviceInfoW(deviceHandle, RIDI_PREPARSEDDATA, NULL, &buffSize);
+        if (FAILED(success))
+        {
+            // that didn't work!
+            auto lastError = GetLastError();
+            err() << "GetRawInputData returned [" << success << "] unexpectedly! GetLastError: [" << lastError << "]"
+                  << std::endl;
+            return;
+        }
+        success       = GetRawInputDeviceInfoW(deviceHandle, RIDI_PREPARSEDDATA, preparsedDataChunk, &buffSize);
+        if (FAILED(success))
+        {
+            // that didn't work!
+            auto lastError = GetLastError();
+            err() << "GetRawInputData returned [" << success << "] unexpectedly! GetLastError: [" << lastError << "]"
+                  << std::endl;
+            return;
+        }
+        PHIDP_PREPARSED_DATA preparsedData = reinterpret_cast<PHIDP_PREPARSED_DATA>(preparsedDataChunk);
+        HIDP_CAPS            hCaps         = {0};
+        success = HidP_GetCaps(preparsedData, &hCaps);
+        if (FAILED(success))
+        {
+            // that didn't work!
+            auto lastError = GetLastError();
+            err() << "HidP_GetCaps returned [" << success << "] unexpectedly! GetLastError: [" << lastError << "]"
+                  << std::endl;
+            return;
+        }
+
+        jstkImpl.m_caps.buttonCount = hCaps.NumberInputButtonCaps;
+        auto axes                   = hCaps.NumberInputValueCaps;
+        for (auto index = 0; index < Joystick::AxisCount && index < axes; index++)
+        {
+            jstkImpl.m_caps.axes[GetAxis(index)] = true;
+        }
     }
 
-    keyDataSize = sizeof(keyData);
+    unsigned int xInputIndex = 0;
+    for (size_t i = 0; i < m_sJoysticks.size(); i++)
+    {
+        auto& jstk = m_sJoysticks.at(i);
```

Comment:
Any specific reason for using `.at(i)` over `[]` operator on a `std::vector<>`? I'm not sure if there's a specific convention within the SFML codebase of when to use one over the other, or if to always prefer one.